src/test/ui/consts/offset_of.rs

```diff
@@ -17,6 +22,15 @@ pub const OFFSET: usize = {
     offset as usize
 };
 
+pub const OFFSET_2: usize = {
+    let uninit = std::mem::MaybeUninit::<Struct2>::uninit();
+    let base_ptr: *const Struct2 = &uninit as *const _ as *const Struct2;
+    let field_ptr = unsafe { &(*base_ptr).field as *const u8 };
```

Comment:
Maybe add a comment (here and above) that this code *is* UB because we are taking references to uninitialized fields.

As a result, it actually might stop compiling any time, if we do more aggressive UB checking during CTFE.